src/pretix/presale/templates/pretixpresale/event/base.html

```diff
@@ -6,6 +6,9 @@
 {% load safelink %}
 {% load rich_text %}
 {% block thetitle %}
+    {% if messages %}
+        {{ messages|join:" " }} :: 
+    {% endif %}
```

Comment:
As you already found out, I took from w3c. Details here: https://www.w3.org/WAI/tutorials/forms/notifications/ 

According to their example, it would probably be better to shorten the messages to sth. like „Error with voucher“ or „Voucher unknown“. But as a first, quick improvement, I placed the whole message. Should we take it out until we have a shortened version?

[Example 4 regarding „Listing errors“](https://www.w3.org/WAI/tutorials/forms/notifications/#listing-errors) is also something to improve upon. I will add both to the ToDo-list.

Reply:
I don't feel we'll add shortened messages anytime soon. We have hundreds of these messages, and shortening them would increase the effort of translating the system into a new language more than it is worth, I believe.

Reply:
If I reed the w3 page correctly, it would be just as good to assign ``role="alert"`` to our ``div class="alert"`` blogs? Or is there is the page title decisively better?

Reply:
`role="alert"` marks a live-region is intended for AJAX-type form-validation/feedback. So AFAICT that doesn’t help here. But we could try to use the `message.tags`and at least output a hint to whether there is an error or warning. Should I look into this?